### PR TITLE
Resolves 519: OnlineIndexer should inherit weak read semantics

### PR DESCRIPTION
**Pull request by MMcM**

### CODE REVIEW

File: docs/ReleaseNotes.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ While not deprecated, the [`MetaDataCache`](https://javadoc.io/page/org.foundati
 * **Bug fix** Fix 4 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
 * **Bug fix** Fix 5 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
 * **Performance** Improvement 1 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
-* **Performance** Improvement 2 [(Issue #NNN)](https://github.com/FoundationDB/fdb-record-layer/issues/NNN)
+* `OnlineIndexer` inherits `WeakReadSemantics` [(Issue #519)](https://github.com/FoundationDB/fdb-record-layer/issues/519)
```
Comment (alecgrieser) [inline docs/ReleaseNotes.md]:
```suggestion
* **Performance** `OnlineIndexer` inherits `WeakReadSemantics` [(Issue #519)](https://github.com/FoundationDB/fdb-record-layer/issues/519)
```

It looks like a similar change was made in the release notes for #530.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java [hunk 2]
```diff
@@ -530,6 +530,24 @@ static void clearMdc(@Nonnull Map<String, String> mdcContext) {
         }
     }
 
+    /**
+     * Get a new {@link FDBDatabaseRunner} that will run contexts similar to this one.
+     * <ul>
+     * <li>Same {@link FDBDatabase database}</li>
+     * <li>Same {@link FDBStoreTimer timer}</li>
+     * <li>Same {@link #getMdcContext() MDC context}</li>
+     * <li>Same {@link FDBDatabase.WeakReadSemantics weak read semantics}</li>
```
Comment (alecgrieser) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java]:
```suggestion
     * <li>Same {@linkplain FDBDatabase.WeakReadSemantics weak read semantics}</li>
```

Same for the other links in this comment.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java [hunk 3]
```diff
@@ -530,6 +530,24 @@ static void clearMdc(@Nonnull Map<String, String> mdcContext) {
...
+     * </ul>
+     * @return a new database runner based on this context
+     */
+    public FDBDatabaseRunner newRunner() {
```
Comment (alecgrieser) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java]: Missing nullability annotation

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java [hunk 4]
```diff
@@ -530,6 +530,24 @@ static void clearMdc(@Nonnull Map<String, String> mdcContext) {
...
+        FDBDatabaseRunner runner = database.newRunner();
+        runner.setTimer(timer);
+        runner.setMdcContext(getMdcContext());
+        runner.setWeakReadSemantics(weakReadSemantics);
+        return runner;
```
Comment (alecgrieser) [inline fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java]: I suppose there probably isn't a good way of making sure that all of the things from the context get copied over to the runner (if more fields are added to a context). Oh, well.

File: docs/ReleaseNotes.md (hunk 1 above)
Comment (MMcM) [reply docs/ReleaseNotes.md]: Fixed.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java (hunk 2 above)
Comment (MMcM) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java]: Fixed.

File: fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java (hunk 3 above)
Comment (MMcM) [reply fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java]: Added.

**Comment by alecgrieser** [comment]
Resolves #519.

**Pull request merged**